core/src/main/java/tc/oc/pgm/stats/StatType.java

```diff
@@ -1,43 +1,87 @@
 package tc.oc.pgm.stats;
 
+import static net.kyori.adventure.text.Component.text;
 import static net.kyori.adventure.text.Component.translatable;
 import static net.kyori.adventure.text.format.NamedTextColor.*;
 import static tc.oc.pgm.stats.StatsMatchModule.damageComponent;
 import static tc.oc.pgm.util.text.NumberComponent.number;
 
 import java.util.Locale;
 import net.kyori.adventure.text.Component;
+import tc.oc.pgm.variables.Variable;
 
-public enum StatType {
-  KILLS,
-  DEATHS,
-  ASSISTS,
-  KILL_STREAK,
-  BEST_KILL_STREAK,
-  KILL_DEATH_RATIO,
-  LONGEST_BOW_SHOT {
-    private final String blocks = key + ".blocks";
+public sealed interface StatType {
+  Component makeNumber(Number number);
+
+  Component buildMessage(Component valueComponent);
+
+  Component conciseComponent(StatHolder stats);
+
+  enum BaseStatType implements StatType {
+    KILLS,
+    DEATHS,
+    ASSISTS,
+    KILL_STREAK,
+    BEST_KILL_STREAK,
+    KILL_DEATH_RATIO,
+    LONGEST_BOW_SHOT {
+      private final String blocks = key + ".blocks";
+
+      @Override
+      public Component makeNumber(Number number) {
+        return translatable(blocks, number(number, YELLOW));
+      }
+    },
+    DAMAGE {
+      @Override
+      public Component makeNumber(Number number) {
+        return damageComponent(number.doubleValue(), GREEN);
+      }
+    };
+
+    public final String key = "match.stats.type." + name().toLowerCase(Locale.ROOT);
 
     @Override
     public Component makeNumber(Number number) {
-      return translatable(blocks, number(number, YELLOW));
+      return number(number, this == DEATHS ? RED : GREEN);
+    }
+
+    @Override
+    public Component conciseComponent(StatHolder stats) {
```

Comment:
I don't see the need for this to change from `component` to `conciseComponent`, it's the same as it was as far as i can tell

Reply:
It is the same functionally, I more or less changed it for semantics since I have a separate interface method for building the stats display component as well. Can change it back to `component` though